lively.ide/world.js

```diff
@@ -1545,33 +1545,47 @@ export class LivelyWorld extends World {
 
 class FileWatcher {
   constructor () {
-    this.fileActions = {};
+    this.filesToWatch = {};
+    this.regexToWatch = {};
     this.onFileSave.bind(this);
     subscribe('file/save', this.onFileSave.bind(this));
   }
 
   clear () {
-    this.fileActions = {};
+    this.filesToWatch = {};
+    this.regexToWatch = {};
   }
 
-  registerFileAction (file, cB) {
-    if (this.fileActions[file]) {
-      if (Array.isArray(this.fileActions[file])) this.fileActions[file].push(cB);
-      else this.fileActions[file] = [this.fileActions[file], cB];
-    } else { this.fileActions[file] = cB; }
+  registerFileAction (fileResourceOrRegex, cB) {
+    if (fileResourceOrRegex.isResource) {
+      const file = fileResourceOrRegex;
+      if (this.filesToWatch[file]) this.filesToWatch[file].push(cB);
+      else this.filesToWatch[file] = [cB];
+      return;
+    }
+
+    fileResourceOrRegex = String(fileResourceOrRegex);
+    if (this.regexToWatch[fileResourceOrRegex]) this.regexToWatch[fileResourceOrRegex].push(cB);
+    else this.regexToWatch[fileResourceOrRegex] = [cB];
   }
 
-  unregisterFileActions (file) {
-    this.fileActions[file] = null;
+  unregisterfilesToWatch (fileResourceOrRegex) {
+    if (fileResourceOrRegex.isResource) this.filesToWatch[fileResourceOrRegex] = null;
+    else this.regexToWatch[String(fileResourceOrRegex)] = null;
   }
 
   async onFileSave (savedFileData) {
-    if (!this.fileActions[savedFileData.resource]) return;
-    const actions = this.fileActions[savedFileData.resource];
-    if (Array.isArray(actions)) {
-      for (let action of actions) {
-        await action();
-      }
-    } else await actions();
+    const actions = this.filesToWatch[savedFileData.resource] || [];
+
+    for (const [k, v] of Object.entries(this.regexToWatch)) {
+      const regex = new RegExp(k);
```

Comment:
This will actually not work: `new RegExp(String(/.*.js/))` will not be equal to `/.*.js/`. Since stringification also includes the enclosing slashes the passed string is not the same regexp any more. Instead you need to rehydrate the regexp like so: 
```js
var parts = regexString.match(/\/(.*?)\/([a-z]*)/);
if (parts) {
  var pattern = parts[1];
  var flags = parts[2];
  var re = new RegExp(pattern, flags);
}
```